core/src/transport/global_only.rs

```diff
@@ -0,0 +1,355 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use crate::{
+    multiaddr::{Multiaddr, Protocol},
+    transport::{ListenerId, TransportError, TransportEvent},
+    Transport,
+};
+use log::debug;
+use std::{
+    net::{Ipv4Addr, Ipv6Addr},
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// Dropping all dial requests to non-global IP addresses.
+#[derive(Debug, Clone, Default)]
+pub struct GlobalIpOnly<T> {
+    inner: T,
+}
+
+pub struct Ipv4Global;
+pub struct Ipv6Global;
+
+impl Ipv4Global {
+    /// Returns [`true`] if this address part of the `0.0.0.0/8` range.
+    pub const fn is_this_network(a: Ipv4Addr) -> bool {
```

Comment:
Unless necessary, I'd prefer for all these to be private.

Reply:
@thomaseizinger Done